WordPress/Classes/Services/BlazeService.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import WordPressKit
+
+final class BlazeService {
+
+    private let contextManager: CoreDataStack
+    private let remote: BlazeServiceRemote
+
+    // MARK: - Init
+
+    required init?(contextManager: CoreDataStack = ContextManager.shared,
+                   remote: BlazeServiceRemote? = nil) {
+        guard let account = try? WPAccount.lookupDefaultWordPressComAccount(in: contextManager.mainContext) else {
+            return nil
+        }
+
+        self.contextManager = contextManager
+        self.remote = remote ?? .init(wordPressComRestApi: account.wordPressComRestV2Api)
+    }
+
+    // MARK: - Methods
+
+    func getStatus(for blog: Blog,
+                   success: @escaping (Bool) -> Void,
+                   failure: @escaping (Error) -> Void) {
+        guard let siteId = blog.dotComID?.intValue else {
+            failure(BlazeServiceError.invalidSiteId)
+            return
+        }
+
+        remote.getStatus(forSiteId: siteId) { result in
+            switch result {
+            case .success(let approved):
+
+                self.contextManager.performAndSave({ context in
+                    guard let blog = Blog.lookup(withObjectID: blog.objectID, in: context) else {
+                        return
```

Comment:
[Q] Should we call the failure handler here?

Reply:
Fixed! [7d002d9](https://github.com/wordpress-mobile/WordPress-iOS/pull/20127/commits/7d002d92bce0b5b3330f244a8256344cda62a312)